From 171ff8faf6a902e71aa9b153397e4e15540625e9 Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Wed, 6 Dec 2017 01:29:37 +0000
Subject: [PATCH] OS-6497 Non-primary NICs of "joyent" branded zones don't work
 with just "addrconf" Reviewed by: Josh Wilsdon <josh@wilsdon.ca> Approved by:
 Josh Wilsdon <josh@wilsdon.ca>

---
 src/vm/node_modules/VM.js | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index 5ee8e9a1..32a0cafe 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -3429,6 +3429,10 @@ function writeZoneNetfiles(payload, log, callback)
                         static_ips);
                 }
 
+                if (addrconf_found) {
+                    replaceFile(zoneroot, '/etc/addrconf.' + n.interface, '');
+                }
+
                 if (n.hasOwnProperty('primary') && !primary_found) {
                     // Only allow one primary network
                     primary_found = true;
@@ -3439,10 +3443,6 @@ function writeZoneNetfiles(payload, log, callback)
                     if (dhcp_found) {
                         replaceFile(zoneroot, '/etc/dhcp.' + n.interface, '');
                     }
-                    if (addrconf_found) {
-                        replaceFile(zoneroot, '/etc/addrconf.' + n.interface,
-                            '');
-                    }
                 }
             });
         }
-- 
2.21.0

